hack/verify-golint.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# Copyright 2014 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# GoFmt apparently is changing @ head...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+cd "${KUBE_ROOT}"
+
+# Use this function when we are ready to start validating all packages.
+find_files() {
+  find . -not \( \
+      \( \
+        -wholename './output' \
+        -o -wholename './_output' \
+        -o -wholename './_gopath' \
+        -o -wholename './release' \
+        -o -wholename './target' \
+        -o -wholename '*/third_party/*' \
+        -o -wholename '*/vendor/*' \
+      \) -prune \
+    \) -name '*.go'
+}
+
+# For now let's check at a package level until they are all turned on.
+packages=(
+	pkg/security/podsecuritypolicy/capabilities
```

Comment:
makes sense to me, I'll update

On Thu, Jun 23, 2016 at 10:09 AM, Daniel Smith <notifications@github.com>
wrote:

> In hack/verify-golint.sh
> <https://github.com/kubernetes/kubernetes/pull/27911#discussion_r68274114>
> :
>
> > +  find . -not \( \
> > +      \( \
> > +        -wholename './output' \
> > +        -o -wholename './_output' \
> > +        -o -wholename './_gopath' \
> > +        -o -wholename './release' \
> > +        -o -wholename './target' \
> > +        -o -wholename '*/third_party/*' \
> > +        -o -wholename '*/vendor/*' \
> > +      \) -prune \
> > +    \) -name '*.go'
> > +}
> > +
> > +# For now let's check at a package level until they are all turned on.
> > +packages=(
> > +	pkg/security/podsecuritypolicy/capabilities
>
> This is going to be a very long list. What do you think about:
>
>    - Storing this in a file
>    - writing something that detects all packages currently passing to
>    update the file
>    - Throws an error if it ever would remove something from the file?
>
> Then this script would just verify that the file is correct.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27911/files/81b16546aaa3abf81e5de34c3e29d8e0bf88e99d#r68274114>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABYNbPEGuHT-7FmISWrop9UV7Rg-fDpjks5qOr23gaJpZM4I8WTQ>
> .
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>